### DISCUSSION TITLE
Alternative ingredients

### DISCUSSION THREAD

**Discussion by KBD20** [TandoorRecipes/recipes#2724] [Ideas]
This is kind of in line with #1962 
Some recipes list 2 ingredients in the same line (x or y), so far I use x and put "or y" in the comment/note area which can get messy.
An idea for this is a substitutes or alternatives in a recipe and have them under the first ingredient but indented and a different background colour and not contribute to the nutritional info etc.
What would work for this and optional ingredients is manually have indenting and "don't contribute to totals" options.

**Comment by vabene1111** [comment]
nice idea, sadly has a lot of implication for example how to caluclate nutritions, what to put on the shopping list and so on, still something I will keep in mind if I find a solution on how to do it.

**Reply by KBD20** [reply]
I guess one workaround is for the user to click the preferred option and that would change calculations, shopping lists etc. maybe visually swapping them.
I'm not sure how much complication that would add though.

A similar thing I've seen is "x or y, or a combo of both", but I can't think of a solution that isn't even more complicated outside of manually adding both to a shopping list and ignoring amounts.

**Reply by Svagtlys** [reply]
I've found myself thinking a substitution feature would be really nice recently, as we use different kinds of milk and butter than we did when we first started using Tandoor. I would expect "substitutes" and the conversion rate to be defined in the food section, and then in the recipes you have a "default" food but can click it (while viewing) to choose a substitute and then the webpage would re-calculate nutrition when you were done. Then you just keep the notes of "or y" and if you decide to sub the ingredient you just use the substitute feature. Combination substitutes would still be tricky, but maybe you could create a separate "food" entry for the combos?

**Reply by vabene1111** [reply]
this sounds very nice and I would still love such a feature but I fear it will add a lot of complexity for the user that needs to be "well hidden". 

For now I am busy with more important stuff but should I decide to make this a prioirty I will come back to this. Would also need to consider shopping and potentially inventory if we ever do this.
